agent.go

```diff
@@ -50,7 +58,8 @@ type Options struct {
 	Rcvhwm              int                 // ZeroMQ socket option of the same name
 	Sndtimeo            int                 // ZeroMQ socket option of the same name
 	Rcvtimeo            int                 // ZeroMQ socket option of the same name
-	Logger              Logger              // Logjam errors are printed using this interface.
+	Logger              Printer             // Logjam errors are printed using this interface.
```

Comment:
Seeing this two so close together first implied some kind of connection to me. Even though the description is pretty clear. What do you think about renaming `Logger` to `ErrorLogger`?

Reply:
The package needs a logger when running tests. If we don't do this or something similar, then users will need to set up the logger in their tests. Which comes with new problems: when and were to do and we would need to document it.

Reply:
I'd like to stay vague. We might want to log more than just errors in the future.